components/scream/src/physics/p3/p3_functions_cldliq_imm_freezing_impl.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef P3_FUNCTIONS_CLDLIQ_IMM_FREEZING_IMPL_HPP
+#define P3_FUNCTIONS_CLDLIQ_IMM_FREEZING_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+/*
+ * Implementation of p3 contact and immersion freezing droplets function.
+ * Clients should NOT #include this file, but include p3_functions.hpp instead.
+ */
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::cldliq_immersion_freezing(const Spack& t, const Spack& lamc,
+                            const Spack& mu_c, const Spack& cdist1,
+                            const Spack& qc_incld, Spack& qcheti, Spack& ncheti)
+{
+  constexpr Scalar qsmall = C::QSMALL;
+  constexpr Scalar AIMM = C::AIMM;
+  constexpr Scalar RainFrze = C::RainFrze;
+  constexpr Scalar ZeroDegC = C::ZeroDegC;
+  constexpr Scalar CONS5 = C::CONS5;
+  constexpr Scalar CONS6 = C::CONS6;
+
+  const auto qc_not_small_and_t_freezing = (qc_incld >= qsmall) &&
+                                           (t <= RainFrze);
+  if (qc_not_small_and_t_freezing.any()) {
+    Spack expAimmDt, inv_lamc3;
+    expAimmDt.set(qc_not_small_and_t_freezing, exp(AIMM * (ZeroDegC-t)));
+    inv_lamc3.set(qc_not_small_and_t_freezing,
+                  (sp(1.0)/lamc) * (sp(1.0)/lamc) * (sp(1.0)/lamc));
```

Comment:
If you have an integer literal, you can just do `1` instead of `sp(1.0)`. The int should be able to be promoted to the scalar type.